x-pack/plugins/secops/public/components/page/manage_query.tsx

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { omit } from 'lodash/fp';
+import React from 'react';
+import { inputsModel } from '../../store';
+
+interface OwnProps {
+  id: string;
+  loading: boolean;
+  refetch: inputsModel.Refetch;
+  setQuery: (params: { id: string; isLoading: boolean; refetch: inputsModel.Refetch }) => void;
+}
+
+export function manageQuery<T>(WrappedComponent: React.ComponentClass<T> | React.ComponentType<T>) {
+  class ManageQuery extends React.PureComponent<OwnProps> {
+    public componentDidUpdate(prevProps: OwnProps) {
```

Comment:
<img width="875" alt="screen shot 2018-12-17 at 12 20 07 pm" src="https://user-images.githubusercontent.com/1151048/50109866-25d7e800-01f6-11e9-9a00-cf2abd706679.png">

Interestingly I am seeing this error in my IDE but not with the linter or compiler. However, if I change the line above to be:

```ts
class ManageQuery extends React.PureComponent<OwnProps & T> {
```

It gets rid of the type error. I think having that makes more sense as well.